tests/brownie/token/tests/test_transferFrom.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python3
+import brownie
+
+
+def test_sender_balance_decreases(accounts, token):
+    sender_balance = token.balanceOf(accounts[0])
+    amount = sender_balance // 4
+
+    token.approve(accounts[1], amount, {'from': accounts[0]})
+    token.transferFrom(accounts[0], accounts[2], amount, {'from': accounts[1]})
+
+    assert token.balanceOf(accounts[0]) == sender_balance - amount
+
+
+def test_receiver_balance_increases(accounts, token):
+    receiver_balance = token.balanceOf(accounts[2])
+    amount = token.balanceOf(accounts[0]) // 4
+
+    token.approve(accounts[1], amount, {'from': accounts[0]})
+    token.transferFrom(accounts[0], accounts[2], amount, {'from': accounts[1]})
+
+    assert token.balanceOf(accounts[2]) == receiver_balance + amount
+
+
+def test_caller_balance_not_affected(accounts, token):
+    caller_balance = token.balanceOf(accounts[1])
+    amount = token.balanceOf(accounts[0])
+
+    token.approve(accounts[1], amount, {'from': accounts[0]})
+    token.transferFrom(accounts[0], accounts[2], amount, {'from': accounts[1]})
+
+    assert token.balanceOf(accounts[1]) == caller_balance
+
+
+def test_caller_approval_affected(accounts, token):
+    approval_amount = token.balanceOf(accounts[0])
+    transfer_amount = approval_amount // 4
+
+    token.approve(accounts[1], approval_amount, {'from': accounts[0]})
+    token.transferFrom(accounts[0], accounts[2], transfer_amount, {'from': accounts[1]})
```

Comment:
It would be good checking this file with any code checker.

Reply:
@apetrovskiy it's not our code. This code copied from Brownie example project